crates/solver-service/src/signature_validation.rs

```diff
@@ -0,0 +1,165 @@
+			network
+				.the_compact_address
+				.as_ref()
+				.ok_or_else(|| APIError::BadRequest {
+					error_type: ApiErrorType::OrderValidationFailed,
+					message: "TheCompact contract not configured".to_string(),
+					details: None,
+				})?;
+
+		// Get contract address for signature validation
+		let contract_address = {
+			let addr = network
+				.input_settler_compact_address
+				.clone()
+				.unwrap_or_else(|| network.input_settler_address.clone());
+			AlloyAddress::from_slice(&addr.0)
+		};
+
+		// Create compact-specific implementations using factory functions
+		let message_hasher = compact::create_message_hasher();
+		let signature_validator = compact::create_signature_validator();
+
+		// 1. Get domain separator from TheCompact contract
+		let domain_separator =
+			get_domain_separator(delivery_service, the_compact_address, origin_chain_id).await?;
+
+		// 2. Compute message hash using interface
+		let struct_hash = message_hasher.compute_message_hash(&intent.order, contract_address)?;
+
+		// 3. Extract signature using interface
+		let signature = signature_validator.extract_signature(&intent.signature);
+
+		// 4. Validate EIP-712 signature using interface
+		let expected_signer = standard_order.user;
+		let is_valid = signature_validator.validate_signature(
+			domain_separator,
+			struct_hash,
+			&signature,
+			expected_signer,
+		)?;
+
+		if !is_valid {
+			return Err(APIError::BadRequest {
+				error_type: ApiErrorType::OrderValidationFailed,
+				message: "Invalid EIP-712 signature".to_string(),
+				details: None,
+			});
+		}
+		Ok(())
+	}
+}
+
+/// Service for managing signature validation across different standards.
+pub struct SignatureValidationService {
+	validators: HashMap<String, Box<dyn OrderSignatureValidator>>,
+}
+
+impl SignatureValidationService {
+	/// Creates a new SignatureValidationService with default validators.
+	pub fn new() -> Self {
+		let mut validators: HashMap<String, Box<dyn OrderSignatureValidator>> = HashMap::new();
+
+		// Register EIP-7683 validator
+		validators.insert("eip7683".to_string(), Box::new(Eip7683SignatureValidator));
+
+		Self { validators }
+	}
+
+	/// Validates a signature using the appropriate validator for the given standard.
+	pub async fn validate_signature(
+		&self,
+		standard: &str,
+		intent: &IntentRequest,
+		networks_config: &NetworksConfig,
+		delivery_service: &Arc<DeliveryService>,
+	) -> Result<(), APIError> {
+		let validator = self
+			.validators
+			.get(standard)
+			.ok_or_else(|| APIError::BadRequest {
+				error_type: ApiErrorType::OrderValidationFailed,
+				message: format!("No signature validator for standard: {}", standard),
+				details: None,
+			})?;
+
+		validator
+			.validate_signature(intent, networks_config, delivery_service)
+			.await
+	}
+
+	/// Checks if signature validation is required for the given lock type.
+	pub fn requires_signature_validation(lock_type: &LockType) -> bool {
+		matches!(lock_type, LockType::ResourceLock)
+	}
```

Comment:
I think we should move this up into the implementation for Eip7683SignatureValidation as it may be different for other types potentially, or not?